unified-runtime/source/adapters/native_cpu/enqueue.cpp

```diff
@@ -222,6 +317,45 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
     });
     rangeStart = rangeEnd;
   }
+#else
+  const IndexT numWG = {numWG0, numWG1, numWG2};
+  IndexT groupsPerThread;
+  size_t dim = 0;
+  for (size_t t = 0; t < 3; t++)
+    groupsPerThread[t] = numWG[t] / numParallelThreads;
```

Comment:
This is confusing, `groupsPerThread` is an array but after initialization, only a single element of that array is used.

More to the point, in #19550, for non-oneTBB, I simplified the splitting across threads to be done over the linear range, rather than over any specific dimension, and that would probably be better with oneTBB as well.